pkg/syncer/syncer_process_test.go

```diff
@@ -0,0 +1,859 @@
+				map[string]string{"deletion.internal.workloads.kcp.dev/us-west1": time.Now().Format(time.RFC3339)},
+				[]string{"workloads.kcp.dev/syncer-us-west1"}),
+			resourceToProcessLogicalClusterName: "root:org:ws",
+			resourceToProcessName:               "theDeployment",
+			workloadClusterName:                 "us-west1",
+			advancedSchedulingEnabled:           true,
+
+			expectError:         false,
+			expectActionsOnFrom: []clienttesting.Action{},
+			expectActionsOnTo: []clienttesting.Action{
+				createNamespaceAction(
+					"",
+					changeUnstructured(
+						toUnstructured(t, namespace("kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "",
+							map[string]string{
+								"state.internal.workloads.kcp.dev/us-west1": "Sync",
+							},
+							map[string]string{
+								"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+							})),
+						removeNilOrEmptyFields,
+					),
+				),
+				deleteDeploymentAction(
+					"theDeployment",
+					"kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973",
+				),
+			},
+		},
+		"SpecSyncer with AdvancedSchedulingFeature deletion, object does not exists downstream": {
+			direction:              SyncDown,
+			upstreamLogicalCluster: "root:org:ws",
+			fromNamespace: namespace("test", "root:org:ws", map[string]string{
+				"state.internal.workloads.kcp.dev/us-west1": "Sync",
+			}, nil),
+			gvr: schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"},
+			toResources: []runtime.Object{
+				namespace("kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "", map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				},
+					map[string]string{
+						"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+					}),
+			},
+			fromResource: deployment("theDeployment", "test", "root:org:ws",
+				map[string]string{"state.internal.workloads.kcp.dev/us-west1": "Sync"},
+				map[string]string{"another.valid.annotation/this": "value",
+					"deletion.internal.workloads.kcp.dev/us-west1": time.Now().Format(time.RFC3339)},
+				[]string{"workloads.kcp.dev/syncer-us-west1"}),
+			resourceToProcessLogicalClusterName: "root:org:ws",
+			resourceToProcessName:               "theDeployment",
+			workloadClusterName:                 "us-west1",
+			advancedSchedulingEnabled:           true,
+
+			expectError: false,
+			expectActionsOnFrom: []clienttesting.Action{
+				getDeploymentAction("theDeployment", "test"),
+				updateDeploymentAction("test",
+					changeUnstructured(
+						toUnstructured(t, changeDeployment(
+							deployment("theDeployment", "test", "root:org:ws", nil, map[string]string{"another.valid.annotation/this": "value"}, nil),
+						),
+						),
+						// TODO(jmprusi): Those next changes do "nothing", it's just for the test to pass
+						//                as the test expects some null fields to be there...
```

Comment:
what does it mean? We are unnecessarily set those zero values in the controller?